neuromation/api/config.py

```diff
@@ -102,15 +113,183 @@ def to_config(self) -> Dict[str, Any]:
 class _Config:
     auth_config: _AuthConfig
     auth_token: _AuthToken
-    cluster_config: _ClusterConfig
     pypi: _PyPIVersion
     url: URL
     cookie_session: _CookieSession
     version: str
+    cluster_name: str
+    clusters: Mapping[str, ClusterConfig]
+
+
+class Config(metaclass=NoPublicConstructor):
+    def __init__(self, core: _Core, path: Path, config_data: _Config) -> None:
+        self._core = core
+        self._path = path
+        self._config_data = config_data
+
+    @property
+    def username(self) -> str:
+        return self._config_data.auth_token.username
+
+    @property
+    def presets(self) -> Mapping[str, Preset]:
+        cluster = self._config_data.clusters[self._config_data.cluster_name]
+        return MappingProxyType(cluster.resource_presets)
+
+    @property
+    def clusters(self) -> Mapping[str, ClusterConfig]:
+        return MappingProxyType(self._config_data.clusters)
+
+    @property
+    def cluster_name(self) -> str:
+        name = self._config_data.cluster_name
+        return name
+
+    async def fetch(self) -> None:
+        server_config = await get_server_config(
+            self._core._session,
+            self._config_data.url,
+            self._config_data.auth_token.token,
+        )
+        if self.cluster_name not in server_config.clusters:
+            raise RuntimeError(
```

Comment:
if a user was removed from a cluster, how would they recover from such a situation?

Reply:
Perhaps user's config is not relevant in this case, the exception text suggests re-login as the last resort.